### PR TITLE
Markdown format updates

### PR DESCRIPTION
**Pull request by guymers**
- Fix syntax error in markdown template
- Only convert text to markdown if html format is being used
- Add a few blank lines for markdown
- Update the overview section so it uses markdown instead of html

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: skydoc/main.py [hunk 1]
```diff
@@ -225,10 +227,29 @@ def main(argv):
                                  macro_doc_extractor.description))
 
   if FLAGS.format == "markdown":
+    # convert documentation from markdown that will be in a html table
+    for ruleset in rulesets:
```
Comment (davidzchen) [inline skydoc/main.py]: Might be cleaner to move the logic of iterating through the rules and rendering markdown into `MarkdownWriter` and `HtmlWriter`, respectively.

File: skydoc/main.py [hunk 2]
```diff
@@ -225,10 +227,29 @@ def main(argv):
                                  macro_doc_extractor.description))
 
   if FLAGS.format == "markdown":
+    # convert documentation from markdown that will be in a html table
```
Comment (davidzchen) [inline skydoc/main.py]: Nit: capitalize "Convert"

File: skydoc/main.py [hunk 3]
```diff
@@ -225,10 +227,29 @@ def main(argv):
...
+        for attribute in r.attributes:
+          attribute.documentation = mistune.markdown(attribute.documentation)
+        for output in r.outputs:
+          output.documentation = mistune.markdown(output.documentation)
+
     markdown_writer = MarkdownWriter(FLAGS.output_dir, FLAGS.output_file,
                                      FLAGS.zip)
     markdown_writer.write(rulesets)
   elif FLAGS.format == "html":
+    # convert possible markdown formatted text to html
```
Comment (davidzchen) [inline skydoc/main.py]: Nit: capitalize "Convert"

**Comment by davidzchen** [comment]
Thanks for your contribution!

Just a few minor comments. Can you also link to gists containing examples of generated Markdown and HTML with this change?

File: skydoc/main.py (hunk 1 above)
Comment (guymers) [inline skydoc/main.py]: I've moved the markdown formatting into the templates.

**Comment by guymers** [comment]
Here is a repository which contains before and after files: https://github.com/guymers/bazel_skydoc_markdown_updates

File: skydoc/main.py (hunk 1 above)
Comment (davidzchen) [inline skydoc/main.py]: Neat!

**Comment by davidzchen** [comment]
Thanks! This change LGTM.

We are currently in the process of changing our tooling for importing changes to Skydoc to the internal repository. I'll merge this once that is done (should be in the next day or so).

**Comment by guymers** [comment]
No worries

**Pull request merged**